### PR TITLE
[RLC] Support binary and form-data content

### PR DESCRIPTION
**Pull request by joheredi**
* Generates the correct types for multi-part body
* Generates correct type for binary request properties "string | Uint8Array"
* Generates correct type for binary response body "Uint8Array"

This PR depends on https://github.com/Azure/azure-sdk-for-js/pull/18753

### CODE REVIEW

File: test/integration/bodyFormDataRest.spec.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      87,
+      111,
+      114,
+      108,
+      100
+    ]);
+
+    const result = await client.path("/formdata/stream/uploadfile").post({
+      contentType: "multipart/form-data",
+      body: { fileContent: content, fileName },
```
Comment (qiaozha) [inline test/integration/bodyFormDataRest.spec.ts:29]: `{ fileContent: content, fileName: fileName }` missed a fileName key ?

File: src/restLevelClient/schemaHelpers.ts [hunk 2]
```diff
@@ -115,14 +115,17 @@ export function primitiveSchemaToType(
...
     case SchemaType.ByteArray:
-    case SchemaType.Binary:
     case SchemaType.Duration:
     case SchemaType.Credential:
     case SchemaType.UnixTime:
     case SchemaType.Uri:
     case SchemaType.Uuid:
     case SchemaType.String:
       return "string";
+    case SchemaType.Binary:
```
Comment (qiaozha) [inline src/restLevelClient/schemaHelpers.ts:125]: just wondering why SchemaType.ByteArray is not mapping to mediaType ?

**Comment by qiaozha** [comment]
The CI is failing

File: test/integration/bodyFormDataRest.spec.ts (hunk 1 above)
Comment (joheredi) [reply test/integration/bodyFormDataRest.spec.ts:29]: If the key has the same name as a variable there is no need to add key value. For example

```typescript
let foo = "foo"

console.log({ foo: foo })
// output:
// {
//  "foo": "foo"
// } 


console.log({ foo})
// output:
// {
//  "foo": "foo"
// } 
```

File: package.json [hunk 3]
```diff
@@ -45,6 +45,7 @@
         "@azure/logger": "^1.0.0",
         "@opentelemetry/api": "^0.10.2",
         "@types/lodash": "^4.14.149",
+        "cd": "^0.3.3",
```
[the remark this answers is not in the document]
Comment (joheredi) [reply package.json:48]: Yeah I'll remove this, not sure why it got in

File: src/restLevelClient/schemaHelpers.ts [hunk 4]
```diff
@@ -115,14 +115,17 @@ export function primitiveSchemaToType(
     case SchemaType.Char:
       return "string";
     case SchemaType.ByteArray:
-    case SchemaType.Binary:
     case SchemaType.Duration:
     case SchemaType.Credential:
```
Comment (sarangan12) [inline src/restLevelClient/schemaHelpers.ts:119]: Should Credential be mapped to string?

File: test/integration/bodyFileRest.spec.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+describe("BodyFile Client", () => {
+  let client: BodyFileRestClient;
+
+  beforeEach("create client", () => {
+    client = BodyFile({ allowInsecureConnection: true });
+  });
+
+  describe("node.js", () => {
+    it("should getFile", async () => {
```
Comment (sarangan12) [inline test/integration/bodyFileRest.spec.ts:12]: Can you add a test case with empty file?

File: test/integration/bodyFormDataRest.spec.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    const response = await client.path("/formdata/stream/uploadfile").put({
+      body: content,
+      binaryResponse: true,
+      contentType: "application/octet-stream"
+    });
+
+    if (response.status !== "200") {
+      const error = `Unexpected error ${response.status}`;
+      assert.fail(error);
+      throw error;
```
Comment (sarangan12) [inline test/integration/bodyFormDataRest.spec.ts:58]: is this throw really required?

**Review by sarangan12 — APPROVED** [review]

File: src/restLevelClient/schemaHelpers.ts (hunk 4 above)
Comment (joheredi) [reply src/restLevelClient/schemaHelpers.ts:119]: That's a goof point. I don't think we have yet any logic for special handling Credentials. We'd need to dig more into that

File: test/integration/bodyFormDataRest.spec.ts (hunk 6 above)
Comment (joheredi) [reply test/integration/bodyFormDataRest.spec.ts:58]: It is somehow for IntelliSense :(
